src/config/searchWorker.ts

```diff
@@ -16,29 +16,147 @@
+            return totalTimeInSec
+        }
+        // Parses time(format:- ex. 4h20m) in second
+        const matchesNumber = duration.match(/[-+]?\d*\.?\d+/g)
+        const matchesChar = duration.match(/[dhms]/g)
+        for (let i = 0; i < matchesNumber.length; i++) {
+            const _unit = matchesChar[i]
+            const _unitVal = +matchesNumber[i]
+            switch (_unit) {
+                case 'd':
+                    totalTimeInSec += _unitVal * 24 * 60 * 60
+                    break
+                case 'h':
+                    totalTimeInSec += _unitVal * 60 * 60
+                    break
+                case 'm':
+                    totalTimeInSec += _unitVal * 60
+                    break
+                default:
+                    totalTimeInSec += _unitVal
+                    break
+            }
+        }
+        return totalTimeInSec
+    }
+
+    const durationComparator = <T extends string>(a: T, b: T, sortOrder: SortingOrder) =>
+        sortOrder === SortingOrder.DESC
+            ? k8sStyledAgeToSeconds(b) - k8sStyledAgeToSeconds(a)
+            : k8sStyledAgeToSeconds(a) - k8sStyledAgeToSeconds(b)
+
+    const propertyComparatorMap = {
+        age: durationComparator,
+        duration: durationComparator,
+        'last schedule': durationComparator,
+        capacity: numberInStringComparator,
+        cpu: numberInStringComparator,
+        memory: numberInStringComparator,
+        window: durationComparator,
+    }
+
+    /**
+     * Dynamically sorts an array of objects based on a specified property and sorting order.
+     * @param  property - The property by which to sort the objects.
+     * @param  sortOrder - The sorting order ('ASC' for ascending, 'DESC' for descending).
+     * @returns A sorting function.
+     */
+    const dynamicSort = (property: string, sortOrder: SortingOrder) => {
+        return (a: Record<string, string | number>, b: Record<string, string | number>) => {
+            const valueA = a[property]
+            const valueB = b[property]
+
+            // Special cases handling where the property is not in sortable format.
+            if (Object.keys(propertyComparatorMap).includes(property)) {
+                return propertyComparatorMap[property](valueA, valueB, sortOrder)
+            }
+
+            // Handling of numbers and if one property is number and the other is string.
+            if (typeof valueA === 'number' || typeof valueB === 'number') {
+                return numberComparatorBySortOrder(
+                    typeof valueA === 'number' ? valueA : 0,
+                    typeof valueB === 'number' ? valueB : 0,
+                    sortOrder,
+                )
+            }
+
+            // Handling of strings and numbers in string type.
+            if (typeof valueA === 'string' && typeof valueB === 'string') {
+                if (!Number.isNaN(Number(valueA)) || !Number.isNaN(Number(valueB))) {
```

Comment:
this would fail for names like: abc, '1'

Reply:
it's handled in next line